### PR TITLE
Extend config examples with log body filtering

### PR DESCRIPTION
**Pull request by mowies**
This PR adds credit card and IBAN filtering for log bodies to the config examples and tests

### CODE REVIEW

**Review by bacherfl — APPROVED** [review]

**Review by odubajDT — APPROVED** [review]

File: config_examples/masking_logbody.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      Authorization: "Api-Token ${env:API_TOKEN}"
+
+processors:
+  transform/cc:
+    error_mode: ignore
+    trace_statements:
+    log_statements:
+      - context: log
+        statements:
+          - replace_pattern(body, "3\\s*[47](\\s*[0-9]){9}((\\s*[0-9]){4})$", "**** $$2")
```
Comment (pirgeo) [inline config_examples/masking_logbody.yaml:19]: Doesnt the `$` at the end mean that this only identifies numbers if the log message ends with the credit card number? I.e.: 
```
2024/11/08 09:54 [paymentservice] This is a log message that contains this credit card number: 2367 8901 2345 6789
```
Would be redacted but for this it wouldn't work:
```
2024/11/08 09:54 [paymentservice] Customer ABC used AMEX 2367 8901 2345 6789 to pay for transaction 12345.
```

File: internal/testbed/integration/filtering_test.go [hunk 2]
```diff
@@ -155,6 +155,163 @@ func TestFilteringCreditCard(t *testing.T) {
 	}
 }
 
+func TestFilteringLogBody(t *testing.T) {
+	logsCCPlain := []string{
+		"card_master_spaces1 2367 8901 2345 6789",
```
Comment (pirgeo) [inline internal/testbed/integration/filtering_test.go:160]: Maybe we can add test cases for `<text> <number> <text>` or just `<number> <text>` (see my other comment)

File: config_examples/masking_logbody.yaml (hunk 1 above)
Comment (mowies) [reply config_examples/masking_logbody.yaml:19]: yes, that's true, I guess I can adapt it a bit to make the regex more general

File: internal/testbed/integration/filtering_test.go (hunk 2 above)
Comment (mowies) [reply internal/testbed/integration/filtering_test.go:160]: done

File: config_examples/masking_logbody.yaml (hunk 1 above)
Comment (mowies) [reply config_examples/masking_logbody.yaml]: done

**Review by pirgeo — APPROVED** [review]

**Pull request merged**